### PR TITLE
chore: Update codeowners to match EM agreed ownership

### PR DESCRIPTION
**Pull request by desi**
## Explanation
We, the MetaMask team (holistically), have recently spent time discussing code ownership around the packages in core. As a team we have aligned on which teams own which packages/controllers within core. This PR is to reflect that alignment. Additionally, in order to not make releasing challenging for the Wallet Framework (who are the stewards of the core repo) we have added this team as codeowners of the manifest file and the changelog file for all packages. 

For a little context being code owners doesn’t mean you are the only ones who work in that code but rather that you will take responsibility for making sure that the code is maintained, healthy, up to date and that others contributing to it do so in a manner that is aligned with our guidelines, processes and principles.

## References
- [Internal google sheets doc](https://docs.google.com/spreadsheets/d/1fukzs7OBZBnswPTLysGaZO7I7uBOGYFpfXEK3Krkn5o/edit?usp=sharing) teams used to align can be found in the google drive
- Additional discussions were in the #mmig-engineering-mgmt channel on slack

## Checklist

- [ n/a ] I've updated the test suite for new or updated code as appropriate
- [ n/a ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ n/a  ] I've highlighted breaking changes using the "BREAKING" category above as appropriate
- [ n/a ] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

**Review by mindofmar — APPROVED** [review]
Product Safety looks good, cc @AugmentedMode for visibility

**Review by gantunesr — DISMISSED** [review]
Approved for accounts

**Review by Mrtenz — APPROVED** [review]

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -3,15 +3,146 @@
...
+/packages/name-controller/package.json      @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+/packages/name-controller/changelog.md      @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+
+## package /packages/network-controller
+/packages/network-controller           @MetaMask/wallet-framework-engineers @MetaMask/metamask-assets
+
+## package /packages/notification-controller
+/packages/notification-controller              @MetaMask/notifications
+/packages/notification-controller/package.json @MetaMask/notifications @MetaMask/wallet-framework-engineers
+/packages/notification-controller/changelog.md @MetaMask/notifications @MetaMask/wallet-framework-engineers
```
Comment (Prithpal-Sooriya) [inline .github/CODEOWNERS:93]: Ideally our team would own all notifications, but I believe this is currently owned by the @MetaMask/snaps-devs

Comment (Prithpal-Sooriya) [reply .github/CODEOWNERS:93]: We have plans in the future to have a consolidation or unification of notifications (orchestration, creation, etc) :)

**Review by Prithpal-Sooriya — DISMISSED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (Prithpal-Sooriya) [reply .github/CODEOWNERS:93]: I don't want this as a blocker, so can leave it for now.

**Review by mathieuartu — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (FrederikBolding) [reply .github/CODEOWNERS:93]: It does feel a bit premature to make this change until we have consolidated notifications more?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -3,15 +3,146 @@
 
 *                                    @MetaMask/engineering
```
Comment (MajorLift) [inline .github/CODEOWNERS:4]: We should add protection for the CODEOWNERS file. Currently (and hypothetically), anyone in `@MetaMask/engineering` could unilaterally alter or remove this ruleset that we've painstakingly put together.

```suggestion
*                                    @MetaMask/engineering
/.github/                            @MetaMask/wallet-framework-engineers 
```

> To protect a repository fully against unauthorized changes, you also need to define an owner for the CODEOWNERS file itself. The most secure method is to define a CODEOWNERS file in the .github directory of the repository and define the repository owner as the owner of either the CODEOWNERS file (/.github/CODEOWNERS @owner_username) or the whole directory (/.github/ @owner_username).

> -- https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-and-branch-protection

File: .github/CODEOWNERS (hunk 1 above)
Comment (MajorLift) [reply .github/CODEOWNERS:93]: Agreed. Could we restore snaps-devs as co-codeowner for the time being?

```suggestion
/packages/notification-controller              @MetaMask/notifications @MetaMask/snaps-devs 
/packages/notification-controller/package.json @MetaMask/notifications @MetaMask/snaps-devs @MetaMask/wallet-framework-engineers
/packages/notification-controller/changelog.md @MetaMask/notifications @MetaMask/snaps-devs @MetaMask/wallet-framework-engineers
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -3,15 +3,146 @@
...
+
+## package /packages/profile-sync-controller
+/packages/profile-sync-controller              @MetaMask/notifications
+/packages/profile-sync-controller/package.json @MetaMask/notifications @MetaMask/wallet-framework-engineers
+/packages/profile-sync-controller/changelog.md @MetaMask/notifications @MetaMask/wallet-framework-engineers
+
+## package /packages/queued-request-controller
+/packages/queued-request-controller               @MetaMask/wallet-api-platform-engineers
+/packages/queued-request-controller/package.json  @MetaMask/wallet-framework-engineers
+/packages/queued-request-controller/changelog.md  @MetaMask/wallet-framework-engineers
```
Comment (MajorLift) [inline .github/CODEOWNERS:125]: Is this intentional? Not sure why we'd want to keep api platform team from independently pushing updates to changelogs and manifests in this package.

```suggestion
/packages/queued-request-controller/package.json  @MetaMask/wallet-api-platform-engineers @MetaMask/wallet-framework-engineers
/packages/queued-request-controller/changelog.md  @MetaMask/wallet-api-platform-engineers @MetaMask/wallet-framework-engineers
```

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -3,15 +3,146 @@
...
+/packages/notification-controller/package.json @MetaMask/notifications @MetaMask/wallet-framework-engineers
+/packages/notification-controller/changelog.md @MetaMask/notifications @MetaMask/wallet-framework-engineers
+
+## package /packages/notification-services-controller
+/packages/notification-services-controller              @MetaMask/notifications
+/packages/notification-services-controller/package.json @MetaMask/notifications @MetaMask/wallet-framework-engineers
+/packages/notification-services-controller/changelog.md @MetaMask/notifications @MetaMask/wallet-framework-engineers
+
+## package /packages/permission-controller
+/packages/permission-controller        @MetaMask/wallet-api-platform-engineers @MetaMask/wallet-framework-engineers
```
Comment (FrederikBolding) [inline .github/CODEOWNERS:101]:
```suggestion
/packages/permission-controller        @MetaMask/wallet-api-platform-engineers @MetaMask/wallet-framework-engineers @MetaMask/snaps-devs
```
I think it makes sense for Snaps to remain a co-owner of the PermissionController. We expect to make more changes to it in the future and are SMEs on the current implementation.

**Review by darkwing — APPROVED** [review]
UX knows very little about the `announcement-controller` but having one to own is the least we can do (?).

**Review by MajorLift — APPROVED** [review]

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -2,16 +2,98 @@
...
+/packages/keyring-controller/package.json         @MetaMask/accounts-engineers @MetaMask/wallet-framework-engineers
+/packages/keyring-controller/CHANGELOG.md         @MetaMask/accounts-engineers @MetaMask/wallet-framework-engineers
+/packages/logging-controller/package.json         @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+/packages/logging-controller/CHANGELOG.md         @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+/packages/message-manager/package.json            @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+/packages/message-manager/CHANGELOG.md            @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+/packages/name-controller/package.json            @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+/packages/name-controller/CHANGELOG.md            @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+/packages/notification-controller/package.json    @MetaMask/notifications @MetaMask/wallet-framework-engineers
+/packages/notification-controller/CHANGELOG.md    @MetaMask/notifications @MetaMask/wallet-framework-engineers
```
Comment (FrederikBolding) [inline .github/CODEOWNERS:85]:
```suggestion
/packages/notification-controller/package.json    @MetaMask/snaps-devs @MetaMask/wallet-framework-engineers
/packages/notification-controller/CHANGELOG.md    @MetaMask/snaps-devs @MetaMask/wallet-framework-engineers
```
Looks like this was missed in the recent iteration!

**Review by MajorLift — APPROVED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Review by DDDDDanica — APPROVED** [review]

**Review by adonesky1 — APPROVED** [review]
LGTM!

**Review by Prithpal-Sooriya — APPROVED** [review]

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**